packages/js/src/ui/context/AppearanceContext.tsx

```diff
@@ -46,6 +56,9 @@ export const AppearanceProvider = (props: AppearanceProviderProps) => {
   const [styleElement, setStyleElement] = createSignal<HTMLStyleElement | null>(null);
   const [elementRules, setElementRules] = createSignal<string[]>([]);
   const [variableRules, setVariableRules] = createSignal<string[]>([]);
+  const themes = createMemo(() =>
+    Array.isArray(props.appearance?.baseTheme) ? props.appearance?.baseTheme || [] : [props.appearance?.baseTheme || {}]
+  );
```

Comment:
✏ **suggestion:** no need for memoization

```
const themes =
    props.appearance?.baseTheme && Array.isArray(props.appearance.baseTheme)
      ? props.appearance.baseTheme
      : [props.appearance.baseTheme];
```

Btw I'm not sure why it's not picking the appearance prop type.

Reply:
Since we're calling `themes()` twice here, the memo is preventing it from executing two times.